android/src/main/java/com/reactnativespokestack/SpokestackAdapter.kt

```diff
@@ -55,10 +55,19 @@ class SpokestackAdapter(sendFunc:(event: String, data: WritableMap) -> Unit):io.
         reactEvent.putString("url", event.ttsResponse.audioUri.toString())
         sendEvent("synthesize", reactEvent)
       }
-      TTSEvent.Type.PLAYBACK_COMPLETE -> {
+      TTSEvent.Type.PLAYBACK_STARTED -> {
+        reactEvent.putBoolean("playing", true)
+        sendEvent("play", reactEvent)
+      }
+      TTSEvent.Type.PLAYBACK_STOPPED -> {
         reactEvent.putBoolean("playing", false)
         sendEvent("play", reactEvent)
       }
+      TTSEvent.Type.PLAYBACK_COMPLETE -> {
+        // Playback is not pause-able in iOS
+        // so this event does not exist there.
```

Comment:
I wonder about the semantics of the events here. Can playback not be interrupted on iOS by the system giving audio focus to another app (e.g. the phone)?

In my mind, if playback can't be paused on iOS, it would be the PLAYBACK_STOPPED event that doesn't exist there, not PLAYBACK_COMPLETE.

Reply:
> Can playback not be interrupted on iOS by the system giving audio focus to another app (e.g. the phone)?

Only if that app took over the audio session. Audio can keep playing in the background as well if the user sets the right category and category options, since AudioSession handling is left to the user. While that is something we can know, it is not something currently tracked with an event. But you may be right that the iOS event is more like stopped than complete.